string w
int16 n
---
string res